core/src/main/java/kafka/server/share/DelayedShareFetch.java

```diff
@@ -153,10 +133,13 @@ public boolean tryComplete() {
             shareFetchData.groupId(), shareFetchData.memberId(),
             shareFetchData.partitionMaxBytes().keySet());
 
-        topicPartitionDataFromTryComplete = acquirablePartitions();
+        Map<TopicIdPartition, FetchRequest.PartitionData> topicPartitionData = acquirablePartitions();
 
-        if (!topicPartitionDataFromTryComplete.isEmpty())
-            return forceComplete();
+        if (!topicPartitionData.isEmpty()) {
+            replicaManagerFetchDataFromTryComplete = replicaManagerFetchData(topicPartitionData, false);
+            if (!replicaManagerFetchDataFromTryComplete.isEmpty())
+                return forceComplete();
+        }
         log.info("Can't acquire records for any partition in the share fetch request for group {}, member {}, " +
```

Comment:
Now we can come to this code path getting no result from `replicaManagerFetchData`. Hence is the log line still correct?

Reply:
You're right... I'll change it to `
log.info("Fetch cannot be completed for the partitions in the share fetch request for group {}, member {}, " +
                "topic partitions {}", shareFetchData.groupId(),
                shareFetchData.memberId(), shareFetchData.partitionMaxBytes().keySet())
`